src/org/thoughtcrime/securesms/FingerprintAuthenticationHandler.java

```diff
@@ -0,0 +1,213 @@
+package org.thoughtcrime.securesms;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.Build;
+import android.os.CancellationSignal;
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyProperties;
+import android.widget.Toast;
+
+import org.thoughtcrime.securesms.crypto.InvalidPassphraseException;
+import org.thoughtcrime.securesms.crypto.MasterSecret;
+import org.thoughtcrime.securesms.crypto.MasterSecretUtil;
+import org.thoughtcrime.securesms.util.FingerprintAuthenticationUtil;
+import org.whispersystems.jobqueue.util.Base64;
+
+import java.io.IOException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.OAEPParameterSpec;
+import javax.crypto.spec.PSource;
+import static android.content.Context.FINGERPRINT_SERVICE;
+
+/**
+ * Created by bryce on 3/9/2018.
+ */
+
+@TargetApi(Build.VERSION_CODES.M)
```

Comment:
What does this do? (curious)

Reply:
Fingerprint was only introduced in marshmallow.  So targetapi forces lint to treat this code as the specified build version instead of the normal target.  It's so we don't have to increase the minsdk of the project to marshmallow to be able to use new features.